code/modules/antagonists/faction/faction.dm

```diff
@@ -0,0 +1,139 @@
+
+/datum/team/faction
+	/// The abbreviation of this family.
+	var/faction_id = "IDK"
+	/// The specific, occupied family member antagonist datum that is used to reach the handler / check objectives, and from which the above properties (sans points) are inherited.
+	var/datum/antagonist/faction/faction_datum
+	/// The current faction objectives.
+	var/datum/faction_objectives/current_objectives
+	/// Additional objective added by players
+	var/additional_objective
+
+/datum/team/faction/roundend_report()
+	var/list/report = list()
+	report += "<span class='header'>[name]:</span>"
+	if(current_objectives.everyone_objective)
+		report += "Objective: [current_objectives.everyone_objective]"
+	else
+		var/assigned_objective = current_objectives.faction_objectives[faction_datum.type]
+		if(assigned_objective || additional_objective)
+			if(assigned_objective)
+				report += "Main Objective: [assigned_objective]"
+			if(additional_objective)
+				report += "[assigned_objective ? "Additional " : ""]Objective: [additional_objective]"
+
+		else
+			report += "This faction had no objectives to complete." // factions can be set no objectives
+
+	return "<div class='panel redborder'>[report.Join("<br>")]</div>"
+
+/datum/team/faction/proc/set_additional_objective(var/new_objective)
+	additional_objective = new_objective
+	for(var/datum/mind/faction_member in members)
+		to_chat(faction_member, "<span class='notice'>A new objective has been assigned: [additional_objective]</span>")
+
+// here are the actual factions
+
+/datum/team/faction/legion
+	faction_id = "LEG"
+
+/datum/antagonist/faction/legion
+	show_in_antagpanel = TRUE
+	name = "Legionarie"
+	roundend_category = "Ceasar's Legion"
+	faction_name = "Ceasar's Legion"
+	faction_id = "LEG"
+	faction_team_type = /datum/team/faction/legion
+
+/datum/team/faction/ncr
+	faction_id = "NCR"
+
+/datum/antagonist/faction/ncr
+	show_in_antagpanel = TRUE
+	name = "NCR Soldier"
+	roundend_category = "The New California Republic"
+	faction_name = "New California Republic Army"
+	faction_id = "NCR"
+	faction_team_type = /datum/team/faction/ncr
+
+/datum/team/faction/bos
+	faction_id = "BOS"
+
+/datum/antagonist/faction/bos
+	show_in_antagpanel = TRUE
+	name = "BOS Member"
+	roundend_category = "The Brotherhood of Steel"
+	faction_name = "Brotherhood of Steel"
+	faction_id = "BOS"
+	faction_team_type = /datum/team/faction/bos
+
+// undecided on these
+/datum/team/faction/tribal
+	faction_id = "TRIB"
+
+/datum/antagonist/faction/tribal
+	show_in_antagpanel = TRUE
+	name = "Tribal"
+	roundend_category = "The tribals"
+	faction_name = "Tribals"
+	faction_id = "TRIB"
+	faction_team_type = /datum/team/faction/tribal
+
+/datum/team/faction/bighorn
+	faction_id = "BH"
+
+/datum/antagonist/faction/bighorn
+	show_in_antagpanel = TRUE
+	name = "Bighorn Citizen"
+	roundend_category = "Bighorn"
+	faction_name = "Bighorn"
+	faction_id = "BH"
```

Comment:
```suggestion
	faction_id = "TOWN"
```

Reply:
FUCK YOU!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! FUCK YOU!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!